packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -776,6 +807,53 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
         return iconClass.concat(additionalClasses).join(' ');
     }
 
+    /**
+     * Add the parent id of the nodes that need tree indent guideline into a set
+     * @param node the tree node.
+     */
+    protected addActiveNodeParent(node: TreeNode): void {
+        if (this.isExpandable(node)) {
+            if (node.children && node.children.length > 0) {
+                this.parentOfActiveIndentGuideline.add(node.id);
+            }
+        } else {
+            const parent = node.parent;
+            if (parent) {
+                this.parentOfActiveIndentGuideline.add(parent.id);
+            }
+        }
+    }
+
+    /**
+     * Render indent for the file tree depending on the depth
+     * @param node the tree node.
+     * @param depth the depth of the tree node.
+     */
+    protected renderIndent(node: TreeNode, depth: number): React.ReactNode {
+
+        const indentDivs: React.ReactNode[] = [];
+        const indentGuideOption = this.corePreferences['workbench.tree.renderIndentGuides'];
+
+        for (let i = 1; i <= depth; i++) {
+            let nodePtr = node;
+            for (let j = depth; j >= i; j--) {
+                if (nodePtr !== undefined && nodePtr.parent !== undefined) {
+                    nodePtr = nodePtr.parent;
+                }
+            }
```

Comment:
Oh, I wasn't understanding the logic here, but I think I do now... Correct me if I am wrong, but it seems like you are creating the left-most indent first, and then move on the right on each iteration. Hence why the nested for-loops.

I think it would be easier to assume we would start with the right-most indent first, not using `indentDivs.push` but [`indentDivs.unshift`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift) instead (since we'll reverse the logic).

Then the loop would be something like the following:

```ts
        let nodePtr = node;
        for (let i = 0; i < depth; i++) {
            nodePtr = nodePtr.parent;
            const needsNodeGuideline = typeof nodePtr !== 'undefined' && this.parentOfActiveIndentGuideline.has(nodePtr.id);
```

Reply:
@marechal-p  Yes, I think it would make more sense to `unshift` in this case. I will update the code.